controllers/jenkins/pipelinerun/pipelinerun_controller.go

```diff
@@ -210,13 +216,12 @@ func (r *Reconciler) deleteJenkinsJobHistory(pipelineRun *prv1alpha3.PipelineRun
 		JenkinsCore: r.JenkinsCore,
 	}
 	jobName := fmt.Sprintf("job/%s/job/%s", pipelineRun.Namespace, pipelineRun.Spec.PipelineRef.Name)
-	if err = jenkinsClient.DeleteHistory(jobName, buildNum); err == nil {
-		pipelineRun.ObjectMeta.Finalizers = sliceutil.RemoveString(pipelineRun.ObjectMeta.Finalizers, func(item string) bool {
-			return item == prv1alpha3.PipelineRunFinalizerName
-		})
-		err = r.Update(context.TODO(), pipelineRun)
-	} else {
-		err = fmt.Errorf("failed to delete Jenkins job: %s, build: %d, error: %v", jobName, buildNum, err)
+	if err = jenkinsClient.DeleteHistory(jobName, buildNum); err != nil {
+		if strings.Contains(err.Error(), "not found resources") {
```

Comment:
I'd prefer to add a TODO here, we will refactor Jenkins client error which could carry the HTTP response code and original error message.